shapely/geos.py

```diff
@@ -869,23 +875,9 @@ class LGEOS360(LGEOS350):
 
     def __init__(self, dll):
         super(LGEOS360, self).__init__(dll)
-        self.methods['minimum_clearance'] = self.GEOSMinimumClearance
-
-
-class LGEOS380(LGEOS360):
```

Comment:
What's the reason for these changes? Was this commited by mistake?

Reply:
This was recently added to Shapely here: https://github.com/Toblerity/Shapely/commit/087e53d3a34e79211f50b34740c938d2690d3e81

So yes, I believe that Niklasarnitz mistakenly copied over a (modified) older version of Shapely on top of this file, undoing all recent changes. Basically a mistake in merging.

Reply:
Oh yes, I am sorry. I kinda just hacked it together myself to get Cura working on macOS Big Sur. I thought I should post my changes here, bc they might help.

Reply:
I guess it would be cool if you revert these changes to the way they should be - so maybe this Pull-Request would be merged soon and we all could start building Cura for MacOS Big Sur.

Reply:
> In the latest macOS (Big Sur), apple changed the way libraries are found and linked.
> In the current version of shapely, nothing works because of this:
> 
> ```
> Traceback (most recent call last):
>   File "<frozen importlib._bootstrap>", line 968, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 957, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 697, in exec_module
>   File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
>   File "/Users/ultimaker/build/4.6/build/inst/lib/python3.5/site-packages/cura/CuraApplication.py", line 51, in <module>
>   File "/Users/ultimaker/build/4.6/build/inst/lib/python3.5/site-packages/cura/Arranging/Arrange.py", line 7, in <module>
>   File "/Users/ultimaker/build/4.6/build/inst/lib/python3.5/site-packages/UM/Math/Polygon.py", line 10, in <module>
>   File "/Users/ultimaker/build/4.6/build/inst/lib/python3.5/site-packages/UM/Math/ShapelyUtil.py", line 6, in <module>
>   File "/Users/ultimaker/build/env/4.6/inst/lib/python3.5/site-packages/shapely/geometry/__init__.py", line 4, in <module>
>   File "/Users/ultimaker/build/env/4.6/inst/lib/python3.5/site-packages/shapely/geometry/base.py", line 17, in <module>
>   File "/Users/ultimaker/build/env/4.6/inst/lib/python3.5/site-packages/shapely/coords.py", line 8, in <module>
>   File "/Users/ultimaker/build/env/4.6/inst/lib/python3.5/site-packages/shapely/geos.py", line 113, in <module>
>   File "/Users/ultimaker/build/env/4.6/inst/lib/python3.5/site-packages/shapely/geos.py", line 56, in load_dll
> OSError: Could not find lib c or load any of its variants [].
> ```
> 
> The change in geos is only because it fixes the error in Cura (see this Issue: [#7970] ([Ultimaker/Cura#7970](https://github.com/Ultimaker/Cura/issues/7970)) over at Cura) and makes it work on Big Sur.
> I thought I would create a PR so you'll see it :)

Just to confirm, having the same issue. By when is this supposed to work or should I downgrade my mac.

Reply:
If you mean in Cura specifically, we've tried to implement a fix in our mac-build-environment that _should_ fix it in the 4.7.0 actual release (but not in the current beta!) Also if your specific issue is Cura, I'm sure the Shapely people'd appreciate it (not to mention us Cura devs) if you posted in the thread linked above, instead of here :-)

If you mean for Shapely in general, no idea! (We hope that someone --which might have to be us, though we lack some of the mac experience-- will look at this before we push out 4.8 so we don't have to keep a hack in our build system :-) )

Reply:
I had this problem while I was using Cartopy. So in general I wanted to know about the shapely so that I can work with Cartopy.
Thanks for your quick reply. :))